src/clnt_vc.c

```diff
@@ -236,11 +236,11 @@ clnt_vc_ncreatef(const int fd,	/* open file descriptor */
 	ct->ct_closeit = false;
 	ct->ct_wait.tv_usec = 0;
 	ct->ct_waitset = false;
-	ct->ct_addr.buf = mem_alloc(raddr->maxlen);
 
-	memcpy(ct->ct_addr.buf, raddr->buf, raddr->len);
-	ct->ct_addr.len = raddr->len;
-	ct->ct_addr.maxlen = raddr->maxlen;
+	if (sizeof(struct sockaddr_storage) >= raddr->len) {
```

Comment:
Something needs to be done, I think, possibly returning error.  In _rpcb_findaddr_timed(), this address buffer is used as a template for allocating a new address buffer, and at least 2 bytes are unconditionally copied into the resulting buffer, which will be size 0 if this check fails.  An alternative may be that the 2 places it's used (clnt_vc_control() and clnt_dg_control()) return error if it's not set?